### ISSUE TITLE
GI probe indirect lighting on gridmaps

### ISSUE THREAD

**Issue by DevMagicLord** [godotengine/godot#18836]
Godot 3.0.2
Win 7 64 bits GeForce 960

Hi there,

I made a scene with an emissive material and Global Illumination, but lighting only throw  on floor, but walls don't receive any bounces lighting.

https://github.com/DevMagicLord/Godot3/blob/master/bugs/Giprobeissue.jpg

While materials without normal map and roughness map receives all indirect lighting.

https://github.com/DevMagicLord/Godot3/blob/master/bugs/giprobeissue2.jpg

Materials with normal map and roughness, metallic maps, needs to receive lighting bounces.

**Comment by Calinou** [comment]
Does your scene use procedural meshes (such as CubeMesh inside a MeshInstance)? From my testing, GIProbe bakes won't affect them.

**Comment by groud** [comment]
Probably related to #18056

**Comment by DevMagicLord** [comment]
I found GiProbe baking works well on standard objects, and emissive objects must also have the flag "used in baked lighting" to true.

But i imported 3D tiles and put "used in baked lighting" to true before saving a meshlib, and baking the Giprobe doesn't bake well on Gridmaps, it's very dark.

On center this is gridmap while others are objects.
https://github.com/DevMagicLord/Godot3/blob/master/bugs/giprobeGridmapsbaking.jpg?raw=true



The issue is about gridmaps.

**Comment by EzraT** [comment]
This happens to me too with Godot 3.0.6 on Linux.
There is a very clear difference to how GIProbe looks on regular MeshInstance nodes, compared to how it looks on a Gridmap.

**Comment by reduz** [comment]
Sorry, without a test scene, figuring out what is going on is impossible. Closing.

**Comment by EzraT** [comment]
@reduz [Reproduction Project](https://github.com/godotengine/godot/files/2591712/gridmap_gi_issue.zip)

**Comment by EzraT** [comment]
Could this one be added to the 3.1 milestone?

**Comment by reduz** [comment]
Actually, it seems to be working, but the emission channel is not strong enough. Lights can compensate this because they have a bake energy value but materials don't. I promise to add this after 3.1 is out, as I will be doing some rewriting to the GIProbe code.

**Comment by KoBeWi** [comment]
Still valid in 3.2.4 beta3
Can't test on master, because GridMap doesn't work (incompatible meshlib?)

**Comment by SoloCodeNet** [comment]
Gi PRobe not working on voxel based model, no texture on the mesh just magica voxel import 
tested with meshinstance directly work fine, add gridmap with same model (left and right) and re-bake
capture : https://ibb.co/pJ8D9Z8

**Comment by Calinou** [comment]
This appears to be fixed in `master` according to my tests in https://github.com/godotengine/godot/issues/56030.

**Comment by EzraT** [comment]
Issue still present in 3.5.beta3.

**Comment by EzraT** [comment]
I did some more testing, and I found something that could hopefully shed some more light on this problem.

To clarify again, this is what the scene in the MRP looks like when using just meshinstances for the hallways:
![image](https://user-images.githubusercontent.com/781121/163173459-528a5da8-1770-4362-a34e-a2eefc239843.png)

Now here's what the same meshes look like in a gridmap, using the exact same settings in the GIProbe:
![image](https://user-images.githubusercontent.com/781121/163173711-718892ec-22a3-411f-8f1d-0e33190193b9.png)

The GIProbe is turned to 16 in both scenes to demonstrate the problem better.

What I discovered is this:
When I turn down the Bias property on the GIProbe node to zero in the scene with only MeshInstances, things start to look very similar to GridMap scene:
![image](https://user-images.githubusercontent.com/781121/163174545-3dccb5ca-2e84-45be-b137-ace103c5b81b.png)

Changing the Bias value in the Gridmap scene also does not result in any different visuals.

Could it be that GIProbe is not applying the Bias value to gridmaps?
